pkg/endpoint/endpoint.go

```diff
@@ -1701,7 +1722,15 @@ func (e *Endpoint) RunMetadataResolver(bwm bandwidth.Manager, resolveMetadata Me
 					}
 					return annotations[bandwidth.EgressBandwidth], nil
 				})
-				e.UpdateLabels(ctx, labels.LabelSourceAny, identityLabels, info, true)
+				// Since 'controllerBaseLabels' were fetched from k8s then we
+				// should only replace the labels that have their source as
+				// 'k8s' otherwise we will risk on replacing other labels that
+				// don't have other sources.
+				source := labels.LabelSourceK8s
+				if len(initialLabels) != 0 {
+					source = labels.LabelSourceAny
```

Comment:
So if the users sets `initialLabels`, then we're deleting any previously set labels not present in set of `initialLabels +  resolveMetadata()` . I think this needs be documented in the go docs, as currently it only talks about the case where `initialLabels` is unset (which is the common case).

Reply:
Yes, does this look good?
```
// If this list is empty, the labels set on the endpoint will be
// replaced by the labels returned 'MetadataResolverCB' by as long their source
// matches the original source.
// If this list is nil and the resolver is unable to retrieve the endpoint
// labels from k8s, the endpoint will be set with the 'init' identity.
```